src/grasshopper/lib/journeys/base_journey.py

```diff
@@ -112,37 +113,73 @@ def _check_for_threshold_parameters_and_set_thresholds(
         self, parameter_key, parameter_value
     ):
         if parameter_key == "thresholds":
-            for raw_trend_name, threshold_less_than_in_ms in parameter_value.items():
-                trend_name, request_type = self._extract_trend_name(raw_trend_name)
-                thresh_object = {
-                    "less_than_in_ms": threshold_less_than_in_ms,
-                    "actual_value_in_ms": None,
-                    "percentile": 0.8,
-                    "succeeded": None,
-                    "http_method": request_type,
-                }
-                if trend_name in self.environment.stats.trends:
-                    self.environment.stats.trends[trend_name]["thresholds"].append(
-                        thresh_object
-                    )
-                else:
-                    self.environment.stats.trends[trend_name] = {
-                        "tags": self.scenario_args.get("tags", {}),
-                        "thresholds": [thresh_object],
+            thresholds_shape = parameter_value
+            if self._verify_thresholds_shape(thresholds_shape):
+                for trend_name, threshold_values in thresholds_shape.items():
+                    thresh_object = {
+                        "less_than_in_ms": int(threshold_values.get("limit")),
+                        "actual_value_in_ms": None,
+                        "percentile": float(threshold_values.get("percentile", 0.9)),
```

Comment:
i think maybe it might be a good idea to make the .9 a constant

Reply:
also, do we have an example anywhere indicating that you can set the percentile key?

Reply:
Changed top use a GrasshopperConstant. And yes there is an example of that in `examples/example_scnearios.YAML`